HW04.Rmd

```diff
@@ -1,126 +1,186 @@
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+#I also needed a couple of additional packages
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(knitr)
+library(forcats)
+
+# read in the data
+gun_deaths <- read.csv("gun_deaths.csv",h=T)
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+```{r month, warning=FALSE}
+#Gun deaths per month
+knitr::kable(gun_deaths %>% count(month), format="markdown",
+             col.names=c("Month","Numer of gun deaths"))
+```
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+#Originally variable was read as an integer, so I changed its class to be read as a factor
+gun_deaths$month <- as.factor(gun_deaths$month)
+
+#Simple barplot with the number of deaths per month
+ggplot(gun_deaths,mapping = aes(x = month)) +
+  geom_bar() +   
+  scale_x_discrete(labels=c("Jan","Feb","Mar","Apr","May","Jun","Jul",
+                            "Aug","Sep","Oct","Nov","Dec"))+
+  theme_bw()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#Barplot with number of deaths by intent (sorted from highest to lowest values)
+#Sorted using the function fct_infreq from the forcats package
+ggplot(gun_deaths,mapping = aes(x = forcats::fct_infreq(intent))) +
+  geom_bar(col="gray30",fill="gray70")+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+#I want the age to be read as a numeric variable.
+gun_deaths$age <- as.numeric(gun_deaths$age)
+
+#Boxplot showing the age of gun death victims by sex:
+ggplot(gun_deaths,mapping = aes(x = sex, y= age)) +
+  geom_boxplot(fill="gray90",width=0.7)+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))
+
+#This is the mean of the age of female gun death victims:
+F_data <- gun_deaths %>% filter(sex=="F")
+mean(F_data$age)
+```
+
+## How many white males with at least a high school education were killed by guns in 2012?
+```{r education}
+#The answer is:
+kable(gun_deaths %>% filter(sex=="M") %>% filter(race=="White") %>% filter(year=="2012") %>% filter(education !="Less than HS")%>% count(), format="markdown")
```

Comment:
Very clever to mix `kable()` with the dplyr functions, it's a very straightforward way of doing things.